### ISSUE TITLE
is error on visual studio 2022 17.10.0 preview  3

### ISSUE THREAD

**Issue by mzy666888** [dotnet/roslynator#1438]
**Product and Version Used**: 
visual studio 2022 17.10.0 preview  3
roslynator 4.12.0

**Steps to Reproduce**:

**Actual Behavior**:
open every project open error
![image](https://github.com/dotnet/roslynator/assets/2587129/1ffbed1d-8c15-4b97-9c45-cd56ea88f732)

disable it (roslynator) don't show this error

**Expected Behavior**:
```csharp
```

**Comment by josefpihrt** [comment]
Hi,

Could you switch VS language to english so I can see what's in the message? Or could you write here what's in the message?

**Comment by KuhnMa-TLS** [comment]
![image](https://github.com/dotnet/roslynator/assets/89680452/25041928-0d25-45ad-8f28-2c10023cac5a)

**Comment by KuhnMa-TLS** [comment]
Stacktrace for 'Asset synchronization':
System.Reflection.TargetInvocationException : Exception has been thrown by the target of an invocation. ---> System.Collections.Generic.KeyNotFoundException : The given key 'Layer' was not present in the dictionary.
   at System.Collections.Immutable.ImmutableDictionary`2.get_Item(TKey key)
   at Microsoft.CodeAnalysis.Host.Mef.LanguageServiceMetadata..ctor(IDictionary`2 data)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target,Object[] arguments,Signature sig,Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr,Binder binder,Object[] parameters,CultureInfo culture)
   at Microsoft.VisualStudio.Composition.MetadataViewClassProvider.CreateProxy(IReadOnlyDictionary`2 metadata,IReadOnlyDictionary`2 defaultValues,Type metadataViewType)
   at Microsoft.VisualStudio.Composition.ExportProvider.<GetExports>g__CreateLazy|81_0[T,TMetadataView](Export export,IMetadataViewProvider metadataViewProvider)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Microsoft.CodeAnalysis.Host.Mef.VisualStudioMefHostServices.<GetExports>b__5_0[TExtension,TMetadata](ExportKey _)
   at System.Collections.Immutable.ImmutableInterlocked.GetOrAdd[TKey,TValue](ImmutableDictionary`2& location,TKey key,Func`2 valueFactory)
   at Microsoft.CodeAnalysis.Host.Mef.VisualStudioMefHostServices.GetExports[TExtension,TMetadata]()
   at Microsoft.CodeAnalysis.Host.Mef.MefWorkspaceServices.GetSupportedLanguages()
   at Microsoft.CodeAnalysis.Serialization.SerializerService..ctor(SolutionServices workspaceServices)
   at Microsoft.CodeAnalysis.Serialization.SerializerService.Factory.CreateService(HostWorkspaceServices workspaceServices)
   at System.Lazy`1.CreateValue()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Lazy`1.get_Value()
   at Microsoft.CodeAnalysis.Host.Mef.MefWorkspaceServices.GetService[TWorkspaceService]()
   at Microsoft.CodeAnalysis.Host.HostWorkspaceServices.GetRequiredService[TWorkspaceService]()
   at async Microsoft.CodeAnalysis.SolutionState.ComputeChecksumsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.SolutionState.GetChecksumAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.SolutionCompilationState.ComputeChecksumsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.SolutionCompilationState.<>c.<-ctor>b__18_0(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.SolutionCompilationState.GetChecksumAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Remote.SolutionAssetStorage.StoreAssetsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Remote.BrokeredServiceConnection`1.TryInvokeAsync[TService](<Unknown Parameters>)

**Comment by KuhnMa-TLS** [comment]
Stacktrace for 'DesignerAttribute discovery':
System.Reflection.TargetInvocationException : Exception has been thrown by the target of an invocation. ---> System.Collections.Generic.KeyNotFoundException : The given key 'Layer' was not present in the dictionary.
   at System.Collections.Immutable.ImmutableDictionary`2.get_Item(TKey key)
   at Microsoft.CodeAnalysis.Host.Mef.LanguageServiceMetadata..ctor(IDictionary`2 data)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target,Object[] arguments,Signature sig,Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr,Binder binder,Object[] parameters,CultureInfo culture)
   at Microsoft.VisualStudio.Composition.MetadataViewClassProvider.CreateProxy(IReadOnlyDictionary`2 metadata,IReadOnlyDictionary`2 defaultValues,Type metadataViewType)
   at Microsoft.VisualStudio.Composition.ExportProvider.<GetExports>g__CreateLazy|81_0[T,TMetadataView](Export export,IMetadataViewProvider metadataViewProvider)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Microsoft.CodeAnalysis.Host.Mef.VisualStudioMefHostServices.<GetExports>b__5_0[TExtension,TMetadata](ExportKey _)
   at System.Collections.Immutable.ImmutableInterlocked.GetOrAdd[TKey,TValue](ImmutableDictionary`2& location,TKey key,Func`2 valueFactory)
   at Microsoft.CodeAnalysis.Host.Mef.VisualStudioMefHostServices.GetExports[TExtension,TMetadata]()
   at Microsoft.CodeAnalysis.Host.Mef.MefWorkspaceServices.GetSupportedLanguages()
   at Microsoft.CodeAnalysis.Serialization.SerializerService..ctor(SolutionServices workspaceServices)
   at Microsoft.CodeAnalysis.Serialization.SerializerService.Factory.CreateService(HostWorkspaceServices workspaceServices)
   at System.Lazy`1.CreateValue()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Lazy`1.get_Value()
   at Microsoft.CodeAnalysis.Host.Mef.MefWorkspaceServices.GetService[TWorkspaceService]()
   at Microsoft.CodeAnalysis.Host.HostWorkspaceServices.GetRequiredService[TWorkspaceService]()
   at async Microsoft.CodeAnalysis.SolutionState.ComputeChecksumsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.SolutionState.GetChecksumAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.SolutionCompilationState.ComputeChecksumsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.SolutionCompilationState.<>c.<-ctor>b__18_0(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.SolutionCompilationState.GetChecksumAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Remote.SolutionAssetStorage.StoreAssetsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Remote.BrokeredServiceConnection`1.TryInvokeAsync[TService](<Unknown Parameters>)

**Comment by josefpihrt** [comment]
@KuhnMa-TLS Thanks!

**Comment by josefpihrt** [comment]
I think it's a duplicate of #1437

**Comment by mzy666888** [comment]
![image](https://github.com/dotnet/roslynator/assets/2587129/b05c1606-f46e-45d5-a320-f1d2f1874ab4)

log file

=====================
2024-04-13 15:01:04
LimitedFunctionality
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.VisualStudio.Composition.MetadataViewClassProvider.CreateProxy(IReadOnlyDictionary`2 metadata, IReadOnlyDictionary`2 defaultValues, Type metadataViewType)
   at Microsoft.VisualStudio.Composition.ExportProvider.<GetExports>g__CreateLazy|81_0[T,TMetadataView](Export export, IMetadataViewProvider metadataViewProvider)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Microsoft.CodeAnalysis.Host.Mef.VisualStudioMefHostServices.<GetExports>b__5_0[TExtension,TMetadata](ExportKey _)
   at System.Collections.Immutable.ImmutableInterlocked.GetOrAdd[TKey,TValue](ImmutableDictionary`2& location, TKey key, Func`2 valueFactory)
   at Microsoft.CodeAnalysis.Host.Mef.VisualStudioMefHostServices.GetExports[TExtension,TMetadata]()
   at Microsoft.CodeAnalysis.Host.Mef.MefLanguageServices..ctor(MefWorkspaceServices workspaceServices, String language)
   at Microsoft.CodeAnalysis.Host.Mef.MefWorkspaceServices.<>c.<GetLanguageServices>b__21_0(String languageName, MefWorkspaceServices self)
   at System.Collections.Immutable.ImmutableInterlocked.GetOrAdd[TKey,TValue,TArg](ImmutableDictionary`2& location, TKey key, Func`3 valueFactory, TArg factoryArgument)
   at Microsoft.CodeAnalysis.Host.Mef.MefWorkspaceServices.GetLanguageServices(String languageName)
   at Microsoft.CodeAnalysis.SolutionState.AddProject(ProjectInfo projectInfo)
   at Microsoft.CodeAnalysis.SolutionCompilationState.AddProject(ProjectInfo projectInfo)
   at Microsoft.CodeAnalysis.Solution.AddProject(ProjectInfo projectInfo)
   at Microsoft.CodeAnalysis.Workspaces.ProjectSystem.ProjectSystemProjectFactory.<>c__DisplayClass35_1.<CreateAndAddToWorkspaceAsync>b__1(Solution oldSolution)
   at Microsoft.CodeAnalysis.Workspace.<>c.<SetCurrentSolutionAsync>b__27_3(Solution oldSolution, ValueTuple`5 data)
   at Microsoft.CodeAnalysis.Workspace.<SetCurrentSolutionAsync>d__29`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.CodeAnalysis.Workspace.<SetCurrentSolutionAsync>d__27.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Roslyn.Utilities.ValueTaskExtensions.VerifyCompleted[T](ValueTask`1 task, String message)
   at Microsoft.CodeAnalysis.Workspaces.ProjectSystem.ProjectSystemProjectFactory.<>c__DisplayClass35_0.<CreateAndAddToWorkspaceAsync>b__0(Workspace w)
   at Microsoft.CodeAnalysis.Workspaces.ProjectSystem.ProjectSystemProjectFactory.<ApplyChangeToWorkspaceAsync>d__41.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.CodeAnalysis.Workspaces.ProjectSystem.ProjectSystemProjectFactory.<CreateAndAddToWorkspaceAsync>d__35.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioProjectFactory.<CreateAndAddToWorkspaceAsync>d__10.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.CPS.CPSProjectFactory.<CreateProjectContextAsync>d__13.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.ProjectSystem.LanguageServices.Workspace.<>c__DisplayClass40_0.<<OnEvaluationUpdateAsync>g__ProcessInitialEvaluationDataAsync|1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.VisualStudio.ProjectSystem.CommonProjectSystemTools.Rethrow(Exception ex)
   at Microsoft.VisualStudio.ProjectSystem.ExceptionFilter.<>c__DisplayClass2_0.<Guard>g__Action|0()
   at GuardMethodClass.GuardMethod(Func`1 , Func`2 , Func`2 )
---> System.Collections.Generic.KeyNotFoundException: The given key 'Layer' was not present in the dictionary.
   at System.Collections.Immutable.ImmutableDictionary`2.get_Item(TKey key)
   at Microsoft.CodeAnalysis.Host.Mef.LanguageServiceMetadata..ctor(IDictionary`2 data)
--- End of inner exception stack trace ---
===================
2024-04-13 15:01:07
LimitedFunctionality
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.VisualStudio.Composition.MetadataViewClassProvider.CreateProxy(IReadOnlyDictionary`2 metadata, IReadOnlyDictionary`2 defaultValues, Type metadataViewType)
   at Microsoft.VisualStudio.Composition.ExportProvider.<GetExports>g__CreateLazy|81_0[T,TMetadataView](Export export, IMetadataViewProvider metadataViewProvider)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Microsoft.CodeAnalysis.Host.Mef.VisualStudioMefHostServices.<GetExports>b__5_0[TExtension,TMetadata](ExportKey _)
   at System.Collections.Immutable.ImmutableInterlocked.GetOrAdd[TKey,TValue](ImmutableDictionary`2& location, TKey key, Func`2 valueFactory)
   at Microsoft.CodeAnalysis.Host.Mef.VisualStudioMefHostServices.GetExports[TExtension,TMetadata]()
   at Microsoft.CodeAnalysis.Host.Mef.MefLanguageServices..ctor(MefWorkspaceServices workspaceServices, String language)
   at Microsoft.CodeAnalysis.Host.Mef.MefWorkspaceServices.<>c.<GetLanguageServices>b__21_0(String languageName, MefWorkspaceServices self)
   at System.Collections.Immutable.ImmutableInterlocked.GetOrAdd[TKey,TValue,TArg](ImmutableDictionary`2& location, TKey key, Func`3 valueFactory, TArg factoryArgument)
   at Microsoft.CodeAnalysis.Host.Mef.MefWorkspaceServices.GetLanguageServices(String languageName)
   at Microsoft.CodeAnalysis.SolutionState.AddProject(ProjectInfo projectInfo)
   at Microsoft.CodeAnalysis.SolutionCompilationState.AddProject(ProjectInfo projectInfo)
   at Microsoft.CodeAnalysis.Solution.AddProject(ProjectInfo projectInfo)
   at Microsoft.CodeAnalysis.Workspaces.ProjectSystem.ProjectSystemProjectFactory.<>c__DisplayClass35_1.<CreateAndAddToWorkspaceAsync>b__1(Solution oldSolution)
   at Microsoft.CodeAnalysis.Workspace.<>c.<SetCurrentSolutionAsync>b__27_3(Solution oldSolution, ValueTuple`5 data)
   at Microsoft.CodeAnalysis.Workspace.<SetCurrentSolutionAsync>d__29`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.CodeAnalysis.Workspace.<SetCurrentSolutionAsync>d__27.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Roslyn.Utilities.ValueTaskExtensions.VerifyCompleted[T](ValueTask`1 task, String message)
   at Microsoft.CodeAnalysis.Workspaces.ProjectSystem.ProjectSystemProjectFactory.<>c__DisplayClass35_0.<CreateAndAddToWorkspaceAsync>b__0(Workspace w)
   at Microsoft.CodeAnalysis.Workspaces.ProjectSystem.ProjectSystemProjectFactory.<ApplyChangeToWorkspaceAsync>d__41.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.CodeAnalysis.Workspaces.ProjectSystem.ProjectSystemProjectFactory.<CreateAndAddToWorkspaceAsync>d__35.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioProjectFactory.<CreateAndAddToWorkspaceAsync>d__10.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.CPS.CPSProjectFactory.<CreateProjectContextAsync>d__13.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.ProjectSystem.LanguageServices.Workspace.<>c__DisplayClass40_0.<<OnEvaluationUpdateAsync>g__ProcessInitialEvaluationDataAsync|1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.VisualStudio.ProjectSystem.CommonProjectSystemTools.Rethrow(Exception ex)
   at Microsoft.VisualStudio.ProjectSystem.ExceptionFilter.<>c__DisplayClass2_0.<Guard>g__Action|0()
   at GuardMethodClass.GuardMethod(Func`1 , Func`2 , Func`2 )
---> System.Collections.Generic.KeyNotFoundException: The given key 'Layer' was not present in the dictionary.
   at System.Collections.Immutable.ImmutableDictionary`2.get_Item(TKey key)
   at Microsoft.CodeAnalysis.Host.Mef.LanguageServiceMetadata..ctor(IDictionary`2 data)
--- End of inner exception stack trace ---
===================

**Comment by bonyjose** [comment]
latest release fixed the issue.
